src/Microsoft.DotNet.XHarness.CLI/Commands/iOS/iOSPackageCommand.cs

```diff
@@ -1,67 +1,300 @@
-                { "name=|n=", "Name of the test application",  v => _arguments.AppPackageName = v},
-                { "output-directory=|o=", "Directory in which the resulting package will be outputted", v => _arguments.OutputDirectory = v},
-                { "working-directory=|w=", "Directory in which the resulting package will be outputted", v => _arguments.WorkingDirectory = v},
-                { "help|h", "Show this message", v => ShowHelp = v != null },
-            };
-        }
-
-        protected override Task<ExitCode> InvokeInternal()
-        {
-            _log.LogDebug($"iOS Package command called:{Environment.NewLine}Application Name = {_arguments.AppPackageName}");
-            _log.LogDebug($"Working Directory:{_arguments.WorkingDirectory}{Environment.NewLine}Output Directory:{_arguments.OutputDirectory}");
-            _log.LogDebug($"Ignore Files Root Directory:{_arguments.IgnoreFilesRootDirectory}{Environment.NewLine}Traits Root Directory:{_arguments.TraitsRootDirectory}");
-            _log.LogDebug($"MTouch Args:{_arguments.MtouchExtraArgs}{Environment.NewLine}Template Type:{Enum.GetName(typeof(TemplateType), _arguments.SelectedTemplateType)}");
-
-            return Task.FromResult(ExitCode.SUCCESS);
-        }
-    }
-}
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.iOS;
+using Microsoft.DotNet.XHarness.iOS.Shared.Execution;
+using Microsoft.DotNet.XHarness.iOS.Shared.Logging;
+using Microsoft.DotNet.XHarness.iOS.Shared.TestImporter.Templates;
+using Microsoft.DotNet.XHarness.iOS.Shared.TestImporter.Templates.Managed;
+using Microsoft.DotNet.XHarness.iOS.Shared.Utilities;
+using Microsoft.DotNet.XHarness.iOS.TestImporter;
+using Microsoft.Extensions.Logging;
+using Mono.Options;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.iOS
+{
+    /// <summary>
+    /// Command that will create the required project generation for the iOS plaform. The command will ensure that all
+    /// required .csproj and src are created. The command is part of the parent CommandSet iOS and exposes similar
+    /// plus extra options to the one that its Android counterpart exposes.
+    /// </summary>
+    internal class iOSPackageCommand : XHarnessCommand
+    {
+        // TODO: Some more parameters we need to make configurable maybe
+        //private const string NugetPath = "/Library/Frameworks/Mono.framework/Versions/Current/Commands/nuget";
+        //private const string MsBuildPath = "/Library/Frameworks/Mono.framework/Versions/Current/bin/msbuild";
```

Comment:
Added issue to keep track of this: https://github.com/dotnet/xharness/issues/104

Reply:
Thank you!